web-components/src/components/datepicker/DatePicker.ts

```diff
@@ -0,0 +1,123 @@
+import "@/components/datepicker/datepicker-calendar/DatePickerCalendar";
+import { addDays, addWeeks, DayFilters, isDayDisabled, now, subtractDays, subtractWeeks } from "@/utils/dateUtils";
+import { customElement, html, internalProperty, LitElement, property, query } from "lit-element";
+import { DateTime } from "luxon";
+import "../input/Input";
+import "../menu-overlay/MenuOverlay";
+import { MenuOverlay } from "../menu-overlay/MenuOverlay";
+
+export namespace DatePicker {}
+
+@customElement("md-datepicker")
+export class DatePicker extends LitElement {
+  @property({ type: Boolean }) shouldCloseOnSelect = true;
+  @property({ attribute: false }) locale: string = now().locale;
+  @property({ attribute: false }) monthFormat = undefined;
+  @property({ type: String }) maxDate: string | undefined = undefined;
+  @property({ type: String }) minDate: string | undefined = undefined;
+  @property({ attribute: false }) selectedDate: DateTime = now();
+  @property({ attribute: false }) focusedDate: DateTime = now();
+  @property({ attribute: false }) filterDate: Function | undefined = undefined;
+  @property({ attribute: false }) onChange: Function | undefined = undefined;
+  @property({ attribute: false }) onMonthChange: Function | undefined = undefined;
+  @property({ attribute: false }) onSelect: Function | undefined = undefined;
```

Comment:
The @internalProperty decorator automatically sets attribute to false; Maybe in your case better using it ?

Reply:
why do we have onAction as props? this looks like a React pattern

Reply:
@yananym True, those are copied over, but do we want to keep the ability to provide callback functions to the Component? I'd be happy to trim them otherwise.